ts/src/base/Exchange.ts

```diff
@@ -3029,10 +3029,19 @@ export default class Exchange {
     async fetchStatus (params = {}) {
         if (this.has['fetchTime']) {
             const time = await this.fetchTime (params);
-            this.status = this.extend (this.status, {
-                'updated': time,
-                'info': time,
-            });
+            try {
+                this.status = this.extend (this.status, {
+                    'status': 'ok',
+                    'updated': time,
+                    'info': time,
+                });
+            } catch (ex) {
+                this.status = this.extend (this.status, {
```

Comment:
It was a bit questionable, but to see from the following point:
- in userland code user checks for status, and the emulated `fetchStatus` , specifically `fetchTime` does not succeed - it should mean the failure of status. let's assume user has wrapped try/catch on top of it, and it means, the 'status' property is being left still with 'ok' status, even though this last call would be failing. So, when `fetchTime` fails, we set status: fail.